pkg/config/remote/service/service.go

```diff
@@ -682,6 +685,10 @@ func (s *CoreAgentService) refresh() error {
 	if err != nil {
 		s.backoffErrorCount = s.backoffPolicy.IncError(s.backoffErrorCount)
 		s.lastUpdateErr = fmt.Errorf("tuf: %v", err)
+		var errDecodeFailed tufclient.ErrDecodeFailed
+		if errors.As(err, &errDecodeFailed) {
+			s.configStatus = state.ConfigStatusExpired
```

Comment:
As mentioned in another comment, it's not just on a refresh of state from the backend that this can fail. We can receive a valid update from the backend, and then be unable to receive a new update, during which the current local repo expires. So you'll have to catch that too in addition to updates just being plain bad during an immediate refresh from the RC backend.